### PR TITLE
[Feature-Variant](Variant Type) support variant type

### PR DESCRIPTION
**Pull request by eldenmoon**
Variant:
It is a type that, type is dynamic and supports writing and querying semi-structured data of any type and any level of nesting.

## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

File: be/src/vec/columns/column_object.h [hunk 1]
```diff
@@ -188,23 +215,68 @@ class ColumnObject final : public COWHelper<IColumn, ColumnObject> {
...
 
-    explicit ColumnObject(bool is_nullable_);
+    explicit ColumnObject(bool is_nullable_, bool create_root = true);
 
     ColumnObject(Subcolumns&& subcolumns_, bool is_nullable_);
 
     ~ColumnObject() override = default;
 
-    bool can_be_inside_nullable() const override { return true; }
+    bool can_be_inside_nullable() const override { return false; }
```
Comment (yiguolei) [inline be/src/vec/columns/column_object.h:234]: 这个函数可能不对，我们现在column const 和 nullable的关系是确定的，而且比较trick，很多地方的设定，很难通过一个函数来保证。

File: be/src/vec/columns/column_string.h [hunk 2]
```diff
@@ -140,15 +145,22 @@ class ColumnString final : public COWHelper<IColumn, ColumnString> {
     }
 
     void insert(const Field& x) override {
-        const String& s = vectorized::get<const String&>(x);
+        StringRef s;
```
Comment (yiguolei) [inline be/src/vec/columns/column_string.h:148]: 这里是不是直接写一个新的ColumnJSONB

File: be/src/vec/data_types/data_type_array.cpp [hunk 3]
```diff
@@ -148,9 +148,9 @@ void DataTypeArray::to_string(const IColumn& column, size_t row_num, BufferWrita
         }
         WhichDataType which(remove_nullable(nested));
         if (which.is_string_or_fixed_string()) {
-            ostr.write("'", 1);
```
Comment (yiguolei) [inline be/src/vec/data_types/data_type_array.cpp:151]: @amorynan  看看啊

**Pull request closed without merging**